gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/reactor/handler/http/AccessPointHttpAcceptor.java

```diff
@@ -0,0 +1,135 @@
+        final EventManager eventManager,
+        final String environmentId,
+        final List<ReactableAccessPoint> initialReactableAccessPoints,
+        final String path,
+        final ReactorHandler reactor,
+        final Collection<String> serverIds
+    ) {
+        super(reactor, eventManager, environmentId);
+        this.path = path;
+        this.serverIds = serverIds;
+        if (initialReactableAccessPoints == null || initialReactableAccessPoints.isEmpty()) {
+            // Initialize with a default acceptor with null accessPointHost
+            var emptyAccessPoint = new ReactableAccessPoint();
+            emptyAccessPoint.setTarget(ReactableAccessPoint.Target.GATEWAY);
+            this.initializeAcceptors(List.of(emptyAccessPoint));
+        } else {
+            this.initializeAcceptors(initialReactableAccessPoints);
+        }
+    }
+
+    @Override
+    protected HttpAcceptor createAcceptor(String accessPointHost) {
+        return new DefaultHttpAcceptor(accessPointHost, path, apiReactor, serverIds);
+    }
+
+    @Override
+    protected ReactableAccessPoint.Target getTarget() {
+        return ReactableAccessPoint.Target.GATEWAY;
+    }
+
+    @Override
+    public String toString() {
+        return acceptors.stream().map(HttpAcceptor::toString).collect(Collectors.joining(","));
+    }
+
+    @Override
+    public String path() {
+        if (!acceptors.isEmpty()) {
+            return acceptors.get(0).path();
+        } else {
+            return path;
+        }
+    }
+
+    @Override
+    public String host() {
+        if (!acceptors.isEmpty()) {
+            return acceptors.get(0).host();
+        } else {
+            return null;
+        }
+    }
+
+    public List<HttpAcceptor> innerHttpsAcceptors() {
+        return acceptors;
+    }
+
+    @Override
+    public int priority() {
+        if (!acceptors.isEmpty()) {
+            return acceptors.get(0).priority();
+        } else {
+            return -1;
+        }
+    }
+
+    @Override
+    public boolean accept(final Request request) {
+        if (acceptors.isEmpty()) {
+            return false;
+        } else if (acceptors.size() == 1) {
+            return acceptors.get(0).accept(request);
+        } else {
+            return acceptors.stream().anyMatch(defaultHttpAcceptor -> defaultHttpAcceptor.accept(request));
+        }
+    }
+
+    @Override
+    public boolean accept(final String host, final String path, final String serverId) {
+        if (acceptors.isEmpty()) {
+            return false;
+        } else if (acceptors.size() == 1) {
+            return acceptors.get(0).accept(host, path, serverId);
+        } else {
+            return acceptors.stream().anyMatch(defaultHttpAcceptor -> defaultHttpAcceptor.accept(host, path, serverId));
+        }
+    }
+
+    @Override
+    public int compareTo(final HttpAcceptor other) {
+        if (!acceptors.isEmpty()) {
+            return acceptors.get(0).compareTo(other);
```

Comment:
I don't really understand why we're only taking the 1st element here, so I've kept the original code. 
but we could use the abstract on 🤷‍♂️